ChemistryLib/PhreeqcInterface.cpp

```diff
@@ -0,0 +1,312 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include <boost/tokenizer.hpp>
+#include <cmath>
+
+#include "BaseLib/ConfigTreeUtil.h"
+#include "Output.h"
+#include "PhreeqcInterface.h"
+#include "ThirdParty/iphreeqc/src/IPhreeqc.h"
+
+namespace ChemistryLib
+{
+void phreeqcIO::doWaterChemistryCalculation(
+    std::vector<GlobalVector*>& process_solutions,
+    ProcessLib::Process const& process, double const dt)
+{
+    setAqueousSolutionsOrUpdateProcessSolutions(
+        process_solutions, process, Status::SettingAqueousSolutions);
+    setTimeStep(dt);
+
+    writeInputsToFile();
+
+    execute();
+
+    readOutputsFromFile();
+
+    setAqueousSolutionsOrUpdateProcessSolutions(
+        process_solutions, process, Status::UpdatingProcessSolutions);
+}
+
+void phreeqcIO::setAqueousSolutionsOrUpdateProcessSolutions(
+    std::vector<GlobalVector*> const& process_solutions,
+    ProcessLib::Process const& process, Status const status)
+{
+    int const nNodes = _solutions.size();
```

Comment:
Just for better understanding of the phreeqc language: a solution is a node in some graph?

Reply:
variable names, please, not in camel-case...

Reply:
the variable `_solutions` collects all the nodal solutions...

Reply:
Fixed.

Reply:
there are more of the camel-case names, like outFile, nNodes, etc. through the whole PR.

Reply:
Fixed and gone through the whole PR. Hope no camel-case name is left...

Reply:
inFile, phreeqcIO -> phreeqc_io

Reply:
Fixed.